### ISSUE TITLE
Intermittent timeout in /eventsource/dedicated-worker/eventsource-prototype.htm and /eventsource/dedicated-worker/eventsource-url.htm

### ISSUE THREAD

**Issue by jdm** [servo/servo#12516]
```
  ▶ TIMEOUT [expected OK] /eventsource/dedicated-worker/eventsource-prototype.htm
  │ 
  │ Xlib:  extension "XFree86-VidModeExtension" missing on display ":0".
  │ thread 'ServiceWorkerManager' panicked at 'Unexpected resource channel panic in sw-manager: RecvError', ../src/libcore/result.rs:785
  │ stack backtrace:
  │   ALERT: RESULT: ["/eventsource/dedicated-worker/eventsource-prototype.htm",0,null,null,[["dedicated worker - EventSource: prototype et al",0,null,null]]]
  │  1:     0x7f08f5bec6af - std::sys::backtrace::tracing::imp::write::h6528da8103c51ab9
  │    2:     0x7f08f5bf42db - std::panicking::default_hook::_$u7b$$u7b$closure$u7d$$u7d$::hbe741a5cc3c49508
  │    3:     0x7f08f5bf3f3f - std::panicking::default_hook::he0146e6a74621cb4
  │    4:     0x7f08f57a7147 - util::panicking::initiate_panic_hook::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::he2b22674ad1748f3
  │    5:     0x7f08f5bda72c - std::panicking::rust_panic_with_hook::h983af77c1a2e581b
  │    6:     0x7f08f5bf4521 - std::panicking::begin_panic::he426e15a3766089a
  │    7:     0x7f08f5bdbf9a - std::panicking::begin_panic_fmt::hdddb415186c241e7
  │    8:     0x7f08f5bf44be - rust_begin_unwind
  │    9:     0x7f08f5c2aacf - core::panicking::panic_fmt::hf4e16cb7f0d41a25
  │   10:     0x7f08f4abe6a2 - core::result::unwrap_failed::h36b4b47aa6131a86
  │   11:     0x7f08f4b8ee28 - script::serviceworker_manager::ServiceWorkerManager::receive_message::h86486463373e871e
  │   12:     0x7f08f4b86347 - std::panicking::try::call::h6801e57145a8dc78
  │   13:     0x7f08f5bfe71b - __rust_try
  │   14:     0x7f08f5bfe6be - __rust_maybe_catch_panic
  │   15:     0x7f08f4b8697f - _<F as alloc..boxed..FnBox<A>>::call_box::ha4ccb389462075c8
  │   16:     0x7f08f5bf26e4 - std::sys::thread::Thread::new::thread_start::h9c883b6d445ece46
  │   17:     0x7f08f17ff183 - start_thread
  │   18:     0x7f08f131637c - clone
  └   19:                0x0 - <unknown>
```
cc @creativcoder. We should fix this ASAP to avoid more intermittent failures.

**Comment by creativcoder** [comment]
Making a PR for that soon.

**Comment by asajeffrey** [comment]
The same root cause?
```
  ▶ TIMEOUT [expected OK] /eventsource/eventsource-request-cancellation.htm
  │ 
  │ Xlib:  extension "XFree86-VidModeExtension" missing on display ":0".
  │ ERROR:servo: Unexpected resource channel panic in sw-manager: RecvError
  │ thread 'ServiceWorkerManager' panicked at 'Unexpected resource channel panic in sw-manager: RecvError', ../src/libcore/result.rs:785
  │ stack backtrace:
  │    1:     0x7f6d3dec3e8f - std::sys::backtrace::tracing::imp::write::h6528da8103c51ab9
  │    2:     0x7f6d3decb99b - std::panicking::default_hook::_$u7b$$u7b$closure$u7d$$u7d$::hbe741a5cc3c49508
  │    3:     0x7f6d3decb60f - std::panicking::default_hook::he0146e6a74621cb4
  │    4:     0x7f6d3deb1f1e - std::panicking::rust_panic_with_hook::h983af77c1a2e581b
  │    5:     0x7f6d3decbbe1 - std::panicking::begin_panic::he426e15a3766089a
  │    6:     0x7f6d3deb377a - std::panicking::begin_panic_fmt::hdddb415186c241e7
  │    7:     0x7f6d3decbb7e - rust_begin_unwind
  │    8:     0x7f6d3df0218f - core::panicking::panic_fmt::hf4e16cb7f0d41a25
  │    9:     0x7f6d3cc4eda2 - core::result::unwrap_failed::h36b4b47aa6131a86
  │   10:     0x7f6d3cd1bef8 - script::serviceworker_manager::ServiceWorkerManager::receive_message::h86486463373e871e
  │   11:     0x7f6d3cd133c7 - std::panicking::try::call::h6801e57145a8dc78
  │   12:     0x7f6d3ded5ddb - __rust_try
  │   13:     0x7f6d3ded5d7e - __rust_maybe_catch_panic
  │   14:     0x7f6d3cd139ff - _<F as alloc..boxed..FnBox<A>>::call_box::ha4ccb389462075c8
  │   15:     0x7f6d3dec9ec4 - std::sys::thread::Thread::new::thread_start::h9c883b6d445ece46
  │   16:     0x7f6d39d1b183 - start_thread
  │   17:     0x7f6d3983237c - clone
  └   18:                0x0 - <unknown>

  ▶ TIMEOUT [expected OK] /eventsource/dedicated-worker/eventsource-constructor-non-same-origin.htm
  │ 
  │ Xlib:  extension "XFree86-VidModeExtension" missing on display ":0".
  │ ERROR:servo: Unexpected resource channel panic in sw-manager: RecvError
  │ thread 'ServiceWorkerManager' panicked at 'Unexpected resource channel panic in sw-manager: RecvError', ../src/libcore/result.rs:785
  │ stack backtrace:
  │    1:     0x7fd233352e8f - std::sys::backtrace::tracing::imp::write::h6528da8103c51ab9
  │    2:     0x7fd23335a99b - std::panicking::default_hook::_$u7b$$u7b$closure$u7d$$u7d$::hbe741a5cc3c49508
  │    3:     0x7fd23335a60f - std::panicking::default_hook::he0146e6a74621cb4
  │    4:     0x7fd233340f1e - std::panicking::rust_panic_with_hook::h983af77c1a2e581b
  │    5:     0x7fd23335abe1 - std::panicking::begin_panic::he426e15a3766089a
  │    6:     0x7fd23334277a - std::panicking::begin_panic_fmt::hdddb415186c241e7
  │    7:     0x7fd23335ab7e - rust_begin_unwind
  │    8:     0x7fd23339118f - core::panicking::panic_fmt::hf4e16cb7f0d41a25
  │    9:     0x7fd2320ddda2 - core::result::unwrap_failed::h36b4b47aa6131a86
  │   10:     0x7fd2321aaef8 - script::serviceworker_manager::ServiceWorkerManager::receive_message::h86486463373e871e
  │   11:     0x7fd2321a23c7 - std::panicking::try::call::h6801e57145a8dc78
  │   12:     0x7fd233364ddb - __rust_try
  │   13:     0x7fd233364d7e - __rust_maybe_catch_panic
  │   14:     0x7fd2321a29ff - _<F as alloc..boxed..FnBox<A>>::call_box::ha4ccb389462075c8
  │   15:     0x7fd233358ec4 - std::sys::thread::Thread::new::thread_start::h9c883b6d445ece46
  │   16:     0x7fd22f1aa183 - start_thread
  │   17:     0x7fd22ecc137c - clone
  └   18:                0x0 - <unknown>

  ▶ TIMEOUT [expected OK] /eventsource/eventsource-constructor-non-same-origin.htm
  │ 
  │ Xlib:  extension "XFree86-VidModeExtension" missing on display ":0".
  │ ERROR:servo: Unexpected resource channel panic in sw-manager: RecvError
  │ thread 'ServiceWorkerManager' panicked at 'Unexpected resource channel panic in sw-manager: RecvError', ../src/libcore/result.rs:785
  │ stack backtrace:
  │    1:     0x7f5374a2ae8f - std::sys::backtrace::tracing::imp::write::h6528da8103c51ab9
  │    2:     0x7f5374a3299b - std::panicking::default_hook::_$u7b$$u7b$closure$u7d$$u7d$::hbe741a5cc3c49508
  │    3:     0x7f5374a3260f - std::panicking::default_hook::he0146e6a74621cb4
  │    4:     0x7f5374a18f1e - std::panicking::rust_panic_with_hook::h983af77c1a2e581b
  │    5:     0x7f5374a32be1 - std::panicking::begin_panic::he426e15a3766089a
  │    6:     0x7f5374a1a77a - std::panicking::begin_panic_fmt::hdddb415186c241e7
  │    7:     0x7f5374a32b7e - rust_begin_unwind
  │    8:     0x7f5374a6918f - core::panicking::panic_fmt::hf4e16cb7f0d41a25
  │    9:     0x7f53737b5da2 - core::result::unwrap_failed::h36b4b47aa6131a86
  │   10:     0x7f5373882ef8 - script::serviceworker_manager::ServiceWorkerManager::receive_message::h86486463373e871e
  │   11:     0x7f537387a3c7 - std::panicking::try::call::h6801e57145a8dc78
  │   12:     0x7f5374a3cddb - __rust_try
  │   13:     0x7f5374a3cd7e - __rust_maybe_catch_panic
  │   14:     0x7f537387a9ff - _<F as alloc..boxed..FnBox<A>>::call_box::ha4ccb389462075c8
  │   15:     0x7f5374a30ec4 - std::sys::thread::Thread::new::thread_start::h9c883b6d445ece46
  │   16:     0x7f5370882183 - start_thread
  │   17:     0x7f537039937c - clone
  └   18:                0x0 - <unknown>
```

**Comment by asajeffrey** [comment]
I think this is blocking #12468, perhaps the thing to do is just to disable `/eventsource/dedicated-worker/eventsource-constructor-non-same-origin.htm` and `/eventsource/eventsource-constructor-non-same-origin.htm`, which are the consistenly failing ones.

**Comment by jdm** [comment]
#12518 should fix it today.

**Comment by timvandermeij** [comment]
Fixed by #12518?
